source/manual/covid-19-services.html.md

```diff
@@ -0,0 +1,315 @@
+- **CD:** Concourse
+- **Logging:** Sentry and Logit
+
+### Application structure
+
+Each application is a standard Rails application with:
+
+- question pages, each has their own controller, view and route
+- a check your answers page
+- a confirmation ("Thank you") page
+- ineligible pages (vulnerable people form only)
+- a privacy page
+
+To run one of the applications locally, see the README in the GitHub repo.
+
+
+## Session and data storage
+
+### Session storage
+
+While the user is filling out the form, we use session storage to
+store the user's data.
+
+For the vulnerable people form, we store session data in an encrypted
+cookie, persisted in the browser and sent back to the server for every
+question.  Cookies have a limit of 4KB, so this approach could cause
+errors if the user submits large inputs.
+
+For the business volunteering form, we store the following for 2
+hours:
+
+- the session id in an encrypted cookie
+- the session data in Redis
+
+### Data storage
+
+For both applications, when the user submits the form on the "Check
+your answers" page, the application writes user data to the database.
+The application only has permission to write items to the database.
+For security and privacy reasons, there's no way to read or change
+already-submitted user data.
+
+Developers should not have access to the production database.
+
+
+## Hosting
+
+### Paas
+
+Both applications are hosted on GOV.UK Platform as a Service (PaaS) in
+the Ireland region.  PaaS provides a scalable hosting platform based
+on CloudFoundry managed by GDS.  You can read more in the [PaaS
+technical documentation][paas-docs].
+
+#### Get access
+
+To get access, ask an organisation manager to invite you to the
+`govuk_development` organisation.  They need to give you a role that
+lets you access both `staging` and `production` spaces and the
+applications inside.
+
+#### Log in
+
+Before you start, [install the CloudFoundry CLI][cf-cli].
+
+1. Log in with `cf login -a api.cloud.service.gov.uk --sso`
+2. Select staging or production with `cf target -s <staging or production>`
+
+Both applications are Rails applications, and use the
+[`ruby_buildpack`][] to build and deploy to PaaS.  You can read more
+about [managing PaaS applications][paas-managing] and [monitoring PaaS
+applications][paas-monitoring].
+
+#### Manage backing services
+
+The business volunteer application uses PaaS's [Redis backing
+service][paas-redis].  The backing service is attached to the
+application by a setting in the `manifest.yml` file in the repository
+root.
+
+The JSON data in the `VCAP_SERVICES` environment variable exposes
+settings and credentials for the backing service automatically.
+
+### AWS
+
+We have two AWS accounts (staging and production) for CDN and data
+storage. They contain DynamoDB, IAM and CloudFront resources,
+which were provisioned by Terraform in the [tech-ops-private
+repository](techops-repo).
+
+To log in to AWS:
+
+1. Install `gds-cli` v2.9.0 or above:
```

Comment:
Ooh, ooh, if we merge #2415 soon we can link to that for "how to install the GDS CLI".

Reply:
Resolved here: https://github.com/alphagov/govuk-developer-docs/pull/2423#discussion_r398664510